appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/ResourceState.java

```diff
@@ -67,6 +69,20 @@ public boolean isBusy() {
 
     public void setBusy(boolean busy) {
         this.busy = busy;
+        if (!busy) {
+            busyException = new TwiceBusyException();
```

Comment:
Can this be controlled by a system property? I believe there is significant overhead in collecting stacktrace at every connection close.